py/tools/py/src/pth.rs

```diff
@@ -48,13 +32,90 @@ impl PthFile {
 
         let mut line = String::new();
         while reader.read_line(&mut line).unwrap() > 0 {
-            let entry = Path::new(prefix).join(Path::new(line.trim()));
+            let entry: PathBuf;
+            if self.prefix.is_some() {
+                entry = Path::new(self.prefix.as_deref().unwrap()).join(Path::new(line.trim()));
+            } else {
+                entry = PathBuf::from(line.trim());
+            }
             line.clear();
-            writeln!(writer, "{}", entry.to_string_lossy())
-                .into_diagnostic()
-                .wrap_err("Unable to write new .pth file entry with prefix")?;
+            if entry.file_name().is_some_and(|x| x == "site-packages") {
+                let src_dir = dest.join(entry).canonicalize().unwrap();
+                create_symlinks(&src_dir, &src_dir, &dest)?;
+            } else {
+                writeln!(writer, "{}", entry.to_string_lossy())
+                    .into_diagnostic()
+                    .wrap_err("Unable to write new .pth file entry")?;
+            }
         }
 
         Ok(())
     }
 }
+
+fn create_symlinks(dir: &Path, root_dir: &Path, dst_dir: &Path) -> miette::Result<()> {
+    // Create this directory at the destination.
+    let tgt_dir = dst_dir.join(dir.strip_prefix(root_dir).unwrap());
+    std::fs::create_dir_all(&tgt_dir)
+        .into_diagnostic()
+        .wrap_err(format!(
+            "unable to create parent directory for symlink: {}",
+            tgt_dir.to_string_lossy()
+        ))?;
+
+    // Recurse.
+    let read_dir = fs::read_dir(dir).into_diagnostic().wrap_err(format!(
+        "unable to read directory {}",
+        dir.to_string_lossy()
+    ))?;
+    for entry in read_dir {
+        let entry = entry.into_diagnostic().wrap_err(format!(
+            "unable to read directory entry {}",
+            dir.to_string_lossy()
+        ))?;
+        let path = entry.path();
+        if path.is_dir() {
+            create_symlinks(&path, root_dir, dst_dir)?;
+        } else if dir != root_dir || entry.file_name() != "__init__.py" {
```

Comment:
following this through, what's the case for special casing `__init__.py` here and below?

Reply:
Ah, I see. I think most folks set `--incompatible_default_to_explicit_init_py` and these files don't get created. Good to be defensive here though imo.